File: Selenium/Activities/Activity22.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity22 {
    public static void main(String[] args) throws InterruptedException {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        Actions builder=new Actions(driver);

        driver.get("https://v1.training-support.net/selenium/popups");
        System.out.println("Title of webpage:" + driver.getTitle());
        WebElement button=driver.findElement(By.xpath("//button[contains(@class,'orange button')]"));
        builder.moveToElement(button).build().perform();
        System.out.println("Tooltip message:"+button.getAttribute("data-tooltip"));

        button.click();
        driver.findElement(By.id("username")).sendKeys("admin");
        driver.findElement(By.id("password")).sendKeys("password");
        driver.findElement(By.xpath("//button[text()='Log in']")).click();
        System.out.println("Login message:"+driver.findElement(By.id("action-confirmation")).getText());
        driver.close();


    }
}

File: Selenium/Activity22.py
from selenium import webdriver
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/popups")
    wait = WebDriverWait(driver, 10)
    print("Title of homepage:" + driver.title)
    builder = ActionChains(driver)
    button = driver.find_element(By.XPATH, "//button[contains(@class,'orange button')]")
    builder.move_to_element(button).perform()
    print("Tooltip message:" + button.get_attribute("data-tooltip"))

    button.click()
    driver.find_element(By.ID, "username").send_keys("admin")
    driver.find_element(By.ID, "password").send_keys("password")
    driver.find_element(By.XPATH, "//button[text()='Log in']").click()
    print("Login message:" + driver.find_element(By.ID, "action-confirmation").text)
    driver.close()
